indexer/src/main.ts

```diff
@@ -0,0 +1,91 @@
+import { TypeormDatabase } from '@subsquid/typeorm-store';
+
+import { processor } from './processor';
+import { config } from './config';
+import { EventsProcessing } from './processing/events.processing';
+import { EventInfo } from './processing/event-info.type';
+import { Block } from '@subsquid/substrate-processor';
+import { NftMarketplaceEventType } from './types/marketplace.events';
+import { EntitiesService } from './processing/entities.service';
+import { getLocalStorage } from './processing/storage/local.storage';
+import { BatchService } from './processing/batch.service';
+
+const NFT_INITIALIZED_CLEAR_TIME = 5 * 60 * 1000; // 5 minutes
+
+function getBlockDate(
+  block: Block<{ block: { timestamp: boolean }; event: { args: boolean } }>,
+) {
+  return new Date(block.header.timestamp ?? new Date().getTime());
+}
+
+let possibleNftInitializedEvents: {
+  payload: string;
+  eventInfo: EventInfo;
+}[] = [];
+
+processor.run(new TypeormDatabase(), async (ctx) => {
+  const localStorage = await getLocalStorage(ctx.store);
+  const entitiesService = new EntitiesService(
+    localStorage,
+    new BatchService(ctx.store),
+    ctx.store,
+  );
+  const processing = new EventsProcessing(entitiesService);
+  const firstBlockDate = getBlockDate(ctx.blocks[0]);
+  console.log(
+    `[main] start processing ${ctx.blocks.length} blocks at ${firstBlockDate}.`,
+  );
+  for (const block of ctx.blocks) {
+    const events = block.events.filter((e) => e.args && e.args.message);
```

Comment:
I'm not sure that there can be a case when this condition isn't true because you subscribed only to `Gear.UserMessageSent` events